accepted/0000-caasp-registration.md

```diff
@@ -0,0 +1,130 @@
+
+The idea is to introduce a new checkbox on this page called "Register the cluster nodes to Uyuni/SUSE Manager".
+If the user does not click on the checkbox, the workflow is unchanged.
+We are going to describe it in the next chapter what happens when the user clicks on the checkbox.
+
+## The new workflow
+
+When the checkbox is selected, a new UI form will appear containing asking for the following information:
+
+- SSH port
+- SSH username with `sudo` rights
+- SSH password
+- Activation Key
+- Proxy
+
+Most experienced readers will recognize that these details are the same asked in the "Boostrap minions" page except for the host (the hosts will be gathered from the `kubeconfig` import call).
+
+**Assumptions**: it is assumed that all the details provided by the user are valid for all nodes of the cluster, which means that all the cluster nodes have the same:
+
+- SSH port
+- `root` SSH access or a user with `sudo` rights
+- password for the user above
+
+After the form has been filled out and the `kubeconfig` import has begun, the usual workflow is followed.
+The additional result for the final user would be that each system listed reached after the `kubeconfig` has been imported will be bootstrapped to Uyuni/SUSE Manager.
+
+## How each cluster node is registered
+
+The SSH details supplied by the user in the workflow will be used to bootstrap each reachable node of the cluster. We want to reuse as much code as possible: a call to `XmlRpcSystemHelper#bootstrap` will be issued for each node of the cluster. For the sake of simplicity and feature-wise, each node will be registered as a Salt minion and not as a salt-ssh minion.
+Scalability-wise:
+
+* From the field, it has been reported that a Kubernetes cluster should not be comprised of more than 100 clusters: managing 100 registrations at the same time should not constitute a performance problem for a healthy Uyuni/SUSE Manager.
+* Theoretically, a cluster [must not have more than 5000 nodes](https://cloud.google.com/solutions/scope-and-size-kubernetes-engine-clusters): the same considerations apply as per the above.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+<!-- Why should we **not** do this?
+
+  * obscure corner cases
+  * will it impact performance?
+  * what other parts of the product will be affected?
+  * will the solution be hard to maintain in the future? -->
+
+Some users already have their automation stack (e.g. puppet or cloud-init) to automatically use the Uyuni/SUSE Manager bootstrap script upon cluster node provisioning.
+In that case, the feature introduced with this RFC is not appealing to this segment of users.
+
+# Alternatives
+[alternatives]: #alternatives
+
+<!-- - What other designs/options have been considered?
+- What is the impact of not doing this? -->
+
+## Bootstrapping by pasting private SSH key
```

Comment:
Yet another alternative would be to use SSH key forwarding:
1. the user forwards to the SUMA server a SSH key that is authorized on all cluster nodes
2. in the bootstrap UI instead of providing a username/password the user can select to a forwarded SSH key
3. once bootstrapping is done, the key forwarding is not needed anymore. The use can terminate the SSH connection that forwarded the key.

Reply:
IIUC, from the point of view of the user, it should be something like:

1. `ssh -A root@susemanager`
2. In Uyuni/SUSE Manager UI Java backend, gather the forward keys for a particular user (in this case `root`) and use them to bootstrap the nodes

I am not sure if 2. is feasible from within Java and an `ssh` user.
Am I missing anything?

Reply:
I haven't tested but maybe forwarded keys can be retrieved using a script with special `sudo` privileges. We already have a few such scripts (e.g. for restarting Suma from the UI).

Reply:
I see that this option might have a lot of additional moving parts compared to the one described in the RFC; for this reason, I would consider the described option in the RFC preferable.